### DISCUSSION TITLE
About future plans on different hardware compatibility

### DISCUSSION THREAD

**Discussion by ahmetakman** [lava-nc/lava#212] [Ideas]
Hello everyone,

If I did not misunderstand lava currently supports running on CPU and GPU. Also provides the necessary software to run on loihi systems. I am curious about whether intel labs have any plans on leveraging lava on different hardware like FPGAs and Intel neurocomputing sticks. 

The purpose of my question is even though it is nice to have CPU and GPU  to simulate SNN systems, those kinds of implementation options might reduce the need for loihi hardware in the early stages for the ones who want to build a mobile system. 

All the best,

**Comment by awintel** [comment]
Hi, 

Lava currently supports CPU/GPU compute resources. Support for Loihi 1 & 2 will be released by end of quarter. As a first meaningful end-to-end example we will provide a deep neural network for video processing.

It is definitely conceivable to support other compute resources like FPGAs or even NCUs. We have not prioritized this work at the moment but are open for suggestions. NCU is certainly easier that FPGA. 

For FPGA support it essentially comes down to implementing the appropriate ProcessModel and extending the compiler to FPGA platforms. FPGA enablement could be a great collaborative community project. The upcoming compiler release end of March should provide a clear structure how such extensions can be made. We don't have a short term need for such a feature ourselves but we are open to providing the necessary support and guidance if anyone in the community would like to pick this up (e.g. after the next workshop in April).

Andreas
